### PR TITLE
fix add graph func is not called bug

### PR DESCRIPTION
**Pull request by shenmishajing**
## Motivation

Fix add graph func is not called bug. See issue #621 

## Modification

- Call visualizer.add_graph after model build.
- Add watch_kwargs param to WandbVisBackend.

## Checklist

4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by HAOCHENYE** [comment]
Will it be more proper to call `add_graph` in `NativeVisualizerHook.before_train`, and this feature can be enabled by setting
`watch_model=True` (default False, not sure will it harm the performance)

**Comment by HAOCHENYE** [comment]
BTW, since `add_graph` has not been used by the downstream repo, could we rename it to a more proper name?

**Comment by shenmishajing** [comment]
> Will it be more proper to call `add_graph` in `NativeVisualizerHook.before_train`, and this feature can be enabled by setting `watch_model=True` (default False, not sure will it harm the performance)

I agree with that. But, does the NativeVisualizerHook have been added to runner now? I do not find it now, and that's why I add it in runner init directly.

**Comment by HAOCHENYE** [comment]
`NativeVisualizerHook` should be a custom hook. If users need to visualize something,  they can inherit or copy `NativeVisualizerHook` to implement custom `VisualizerHook`. The code snippet of config to enable `VisualizerHook` is like this:

`config_for_xxx.py`
```python
custom_hooks = [
    dict(type='NativeVisualizerHook')
    ...
]
```

**Comment by shenmishajing** [comment]
> `NativeVisualizerHook` should be a custom hook. If users need to visualize something, they can inherit or copy `NativeVisualizerHook` to implement custom `VisualizerHook`. The code snippet of config to enable `VisualizerHook` is like this:
> 
> `config_for_xxx.py`
> 
> ```python
> custom_hooks = [
>     dict(type='NativeVisualizerHook')
>     ...
> ]
> ```

Ok, got it. I think it's more proper to call add_graph in `NativeVisualizerHook.before_train`.

**Comment by HAOCHENYE** [comment]
I think `add_config` is a more commonly used function for the general experiments, it is ok to be called in `runner.__init__() `.

**Comment by shenmishajing** [comment]
> 

Ok, thanks for your reply.

**Review by hhaAndroid — APPROVED** [review]

File: mmengine/visualization/vis_backend.py [hunk 1]
```diff
@@ -360,19 +360,22 @@ class WandbVisBackend(BaseVisBackend):
             `wandb docs <https://docs.wandb.ai/ref/python/run#log_code>`_
             for details. Defaults to None.
             New in version 0.3.0.
+        watch_kwargs (optional, dict): Agurments for ``wandb.watch``.
```
Comment (zhouzaida) [inline mmengine/visualization/vis_backend.py:363]:
```suggestion
        watch_kwargs (optional, dict): Agurments for ``wandb.watch``.
            New in version 0.3.0.
```

**Comment by zhouzaida** [comment]
Hi @shenmishajing , thanks for your contribution. After having a discussion with WandB team, they have no plan to support visualizing models in `wanbd.watch` but we will merge this PR because it is a useful feature.

File: mmengine/visualization/vis_backend.py [hunk 2]
```diff
@@ -360,19 +360,23 @@ class WandbVisBackend(BaseVisBackend):
...
+            New in version 0.3.0.
```
Comment (zhouzaida) [inline mmengine/visualization/vis_backend.py:364]:
```suggestion
            New in version 0.4.0.
```

**Review by zhouzaida — APPROVED** [review]

**Review by HAOCHENYE — APPROVED** [review]

**Pull request merged**